hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/AbstractTableDetailsLayout.java

```diff
@@ -129,10 +130,13 @@ protected void setName(final String headerCaption, final String value) {
         caption.setValue(HawkbitCommonUtil.getSoftwareModuleName(headerCaption, value));
     }
 
+    /**
+     * Restores the tables and tabs displayed on the view based on the selected
+     * entity.
+     */
     protected void restoreState() {
         if (onLoadIsTableRowSelected()) {
-            populateData(null);
-            editButton.setEnabled(true);
+            populateData(getSelectedBaseEntity());
```

Comment:
You don't need the if (onLoad....) at all. The data should always be populated with the selectedEntity. If nothing is selected than it is null. I remove that.